### PR TITLE
Fixes broken links

### PR DESCRIPTION
**Pull request by athityakumar**
Requires a careful view to check if there any more broken links. A `curl` shell script maybe to loop through all the `Link`?

### CODE REVIEW

**Comment by icyflame** [comment]
@athityakumar Actually, there's a slight problem with diffs like these. So, a good way to ensure that reviewers can see a readable diff is to follow this protocol:

1. pretty print the data.json file, commit. (c8f4e0586677620691a72dcfc845edc11bd36972)
2. Make your changes, and keep the JSON pretty printed. Commit. (b1e28d30c52cb5b87e11507033bf9eefdc4e4ada)
3. And finally, minify the JSON again and commit. (4c04071de1d302103fd28a060479ecdc165f8b31)

So, basically, the diff of commit number 2 in each series following the above protocol will be readable. This is not ideal, and I would like to implement something like #16 to remove this kind of a work around.

I am closing this PR for now, please re-open after pushing the new set of three or more commits.

**Comment by athityakumar** [comment]
@icyflame - I wrote a small ruby script with `Mechanize` to check if there are any broken links, and have fixed them. That way, all broken fields have been fixed manually - I've rechecked twice. Should I also include this Ruby script that I used to identify broken links?

```
require 'mechanize'
require 'json'
agent = Mechanize.new
data = JSON.parse(File.read('data/data.json'))
i = 0
data.each do |d|
    begin
        agent.get(d['Link'])
    rescue
        puts "Check lines : #{(2+i*7).to_s} - #{(8+i*7).to_s}"
    end
    i = i+1
end
```

**Comment by athityakumar** [comment]
Ping @icyflame ^

**Comment by icyflame** [comment]
@athityakumar I think you missed this: https://github.com/metakgp/mfqp/pull/26#issuecomment-286201399

**Comment by athityakumar** [comment]
@icyflame - Yeah, I saw that comment. And I just wanted to suggest a work-around for that commit system.

Instead of those 3 commits to review the PRs by looking at the diffs, can't we just have a Travis CI that checks for broken links with such a ruby script?

```
require 'mechanize'
require 'json'
agent = Mechanize.new
data = JSON.parse(File.read('data/data.json'))
i = 0
data.each do |d|
    begin
        agent.get(d['Link'])
    rescue
        puts "Check lines : #{(2+i*7).to_s} - #{(8+i*7).to_s}"
    end
    i = i+1
end
```

**Comment by icyflame** [comment]
@athityakumar I don't understand, do you want this PR to be merged or not?

If not, then we can close this and continue the discussion on Slack.
If yes, then we _must_ make the diff for one of the commits in this PR human-readable.

Travis CI integration is a separate issue and must be discussed on a new issue. As it is, in your script I can notice two problems:

1. It doesn't check _what_ was the error the agent faced. For eg, if it was a 503 that means the server is down and we can't conclude whether the link is broken or not. The only case where we can conclude that the link is broken is a 404. Also, some other 4xx statuses can have a valid conclusion.

2. This won't help us a lot because we would have to keep checking travis. There are _some_ plugins which open issues on Github depending on whether they find an issue or not. So, my suggested way to move forward with this is:

    - Fix **all** broken links in the present JSON
    - Integrate Travis CI, and enable it to open issues here if a commit introduces broken links into the JSON.

When you open a new issue for Travis CI integration, please copy the second part of the above write-up there instead of linking so as to keep the discussion easy to follow.

**Comment by athityakumar** [comment]
@icyflame - Peace, will open a new issue for Travis CI. Meanwhile, I'also make the 3 commits for a human-readable diff.

**Comment by icyflame** [comment]
ping @athityakumar (please reopen when you can make the diff human readable, thanks!)
